common/src/main/java/org/jboss/pnc/bacon/common/OTELHelper.java

```diff
@@ -0,0 +1,79 @@
+package org.jboss.pnc.bacon.common;
+
+import com.redhat.resilience.otel.internal.EnvarExtractingPropagator;
+import io.opentelemetry.api.common.Attributes;
+import io.opentelemetry.api.trace.Span;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.context.propagation.ContextPropagators;
+import io.opentelemetry.sdk.OpenTelemetrySdk;
+import io.opentelemetry.sdk.resources.Resource;
+import io.opentelemetry.sdk.trace.SdkTracerProvider;
+import io.opentelemetry.sdk.trace.SpanProcessor;
+import io.opentelemetry.semconv.resource.attributes.ResourceAttributes;
+import lombok.experimental.UtilityClass;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+@UtilityClass
+public class OTELHelper {
+    private SpanProcessor processor = null;
+
+    private Span root = null;
+
+    public Span startOtel(String serviceName, String commandName, SpanProcessor p) {
+        processor = p;
+        if (serviceName == null) {
+            serviceName = "bacon";
+        }
+        if (commandName == null) {
+            commandName = serviceName;
+        }
+
+        Resource resource = Resource.getDefault()
+                .merge(Resource.create(Attributes.of(ResourceAttributes.SERVICE_NAME, serviceName)));
+
+        SdkTracerProvider sdkTracerProvider = SdkTracerProvider.builder()
+                .addSpanProcessor(processor)
+                .setResource(resource)
+                .build();
+
+        // NOTE the use of EnvarExtractingPropagator here OpenTelemetrySdk openTelemetry = OpenTelemetrySdk.builder()
+        OpenTelemetrySdk openTelemetry = OpenTelemetrySdk.builder()
+                .setTracerProvider(sdkTracerProvider)
+                .setPropagators(ContextPropagators.create(EnvarExtractingPropagator.getInstance()))
+                .buildAndRegisterGlobal();
+
+        Context parentContext = EnvarExtractingPropagator.getInstance().extract(Context.current(), null, null);
+        log.error("### Parent context {} and {}", parentContext, Context.current().getClass().getName());
+        root = openTelemetry.getTracer(serviceName)
+                .spanBuilder(commandName)
+                .setParent(parentContext)
+                // .setSpanKind(SpanKind.CLIENT)
+                .startSpan();
+
+        // Context.current().
+        root.makeCurrent();
+        log.error(
+                "### Current context traceId {} spanId {}",
+                Span.current().getSpanContext().getTraceId(),
+                Span.current().getSpanContext().getSpanId());
+
+        return root;
+    }
+
+    public boolean otelEnabled() {
+        return processor != null;
+    }
+
+    public void stopOtel() {
+        if (root != null && processor != null) {
+            log.debug("Finishing OTEL instrumentation for {}", root);
+            root.end();
+            processor.close();
+        }
+    }
+
+    public String generateTraceParent(String trace, String span) {
+        return "00" + '-' + trace + '-' + span + "-01";
```

Comment:
Ok this last "-01" might vary (depends mainly if the spans are sampled or not). I don't have a logic for that yet, but we might need to change that in the future, for now we can consider it ok!

Reply:
Sure that makes sense.